pkgs/applications/video/animdl/default.nix

```diff
@@ -0,0 +1,28 @@
+{ fetchFromGitHub, poetry2nix, lib }:
+poetry2nix.mkPoetryApplication rec {
```

Comment:
This should use `python3.pkgs.buildPythonApplication` instead of poetry2nix.

Reply:
The Animdl build instructions said to use poetry so I just used this as it seemed the easiest. I can try doing it that way if needed

Reply:
Visit [nixos.org/manual/nixpkgs/unstable](https://nixos.org/manual/nixpkgs/unstable/) and search for

> buildPythonApplication function

Reply:
Set `format = "pyproject"` and put `poetry-core` into `nativeBuildInputs`. The python builder does support that.

Reply:
I've got pycryptodomex in the attribute set and in the nativeBuildInputs but it complains that it could not find a version that is satisfactory: pycryptodomex<3.15.0,>=3.14.1.
I tried overwriting but that did not seem to work.

Reply:
Try relaxing that requirement in `postPatch` using `substituteInPlace`. Search `pkgs/development/python-modules` for examples-

Reply:
I have it successfully built. I had to pack 2 of its python dependencies so I have a question: What do I do with those? Do I pack them with this or make a PR for those?

Reply:
They go individually into `pkgs/development/python-modules/` and are referenced from `pkgs/top-level/python-modules.nix`.

1. Yes, one commit per attribute
2. Bug the upstream to add one, otherwise it would need to be marked unfree